GameServer.cpp

```diff
@@ -172,9 +172,6 @@ int GameServer::AllocSocket(SOCKET& hostSocket, unsigned short port)
 
 void GameServer::ProcessPacket(Packet &packet, sockaddr_in &from)
 {
-	unsigned int gameInfoIndex;

```

Comment:
When splitting the function, I noticed gameInfoIndex is defined as unsigned, but is being checked against a value of -1. I think the unsigned qualifier needs to be dropped. However, it becomes a large changeset, so would like to solve in another branch to keep this one from ballooning in proportion.

Reply:
It's an index value, so `std::size_t` is probably most appropriate. The `-1` sentinel value could be replaced by `numeric_limits<T>::max()`.

Assuming it's never serialized into a fixed size field in a packet anyway. In that case, a `std::uint32_t` would probably be better.